src/theory/quantifiers/ceg_t_instantiator.cpp

```diff
@@ -1025,110 +1025,82 @@ Node BvInstantiator::hasProcessAssertion(CegInstantiator* ci,
-    Assert(!tm.isNull() && tm.isConst());
-    Trace("cegqi-bv") << "Model value: " << std::endl;
-    Trace("cegqi-bv") << "   " << s << " " << k << " " << t << " is "
-                      << std::endl;
-    Trace("cegqi-bv") << "   " << sm << " <> " << tm << std::endl;
-
-    if (options::cbqiBvIneqMode() == CBQI_BV_INEQ_EQ_SLACK)
+    else
     {
-      useSlack = true;
+      ret = s.eqNode(t);
     }
-    else if (k == EQUAL)
+  } else {
+    // turn disequality into an inequality
+    // e.g. s != t becomes s < t or t < s
+    if (k == EQUAL)
     {
-      Node comp = nm->mkNode(BITVECTOR_ULT, sm, tm);
-      comp = Rewriter::rewrite(comp);
-      k = BITVECTOR_ULT;
-      // go in the direction of the model
-      if (!comp.getConst<bool>())
+      if (Random::getRandom().pickWithProb(0.5))
       {
-        Node u = s;
-        s = t;
-        t = u;
+        std::swap(s, t);
       }
       pol = true;
     }
-
-    // if using slack, we convert constraints to a positive equality based on
-    // the current model M, e.g.:
-    //   (not) s ~ t  --->  s = t + ( s^M - t^M )
-    Node ret;
-    if (useSlack) {
-      if (sm != tm)
-      {
-        Node slack = Rewriter::rewrite(nm->mkNode(kind::BITVECTOR_SUB, sm, tm));
-        Assert(slack.isConst());
-        // remember the slack value for the asserted literal
-        d_alit_to_model_slack[lit] = slack;
-        ret = nm->mkNode(
-            kind::EQUAL, s, nm->mkNode(kind::BITVECTOR_PLUS, t, slack));
-        Trace("cegqi-bv") << "Slack is " << slack << std::endl;
-      }
-      else
-      {
-        ret = s.eqNode(t);
-      }
+    // otherwise, we optimistically solve for the boundary point of an
+    // inequality, for example:
+    //   for s < t, we solve s+1 = t
+    //   for ~( s < t ), we solve s = t
+    // notice that this equality does not necessarily hold in the model, and
+    // hence the corresponding instantiation strategy is not guaranteed to be
+    // monotonic.
+    if (!pol)
+    {
+      ret = s.eqNode(t);
     } else {
-      // otherwise, we optimistically solve for the boundary point of an
-      // inequality, for example:
-      //   for s < t, we solve s+1 = t
-      //   for ~( s < t ), we solve s = t
-      // notice that this equality does not necessarily hold in the model, and
-      // hence the corresponding instantiation strategy is not guaranteed to be
-      // monotonic.
-      if (!pol) {
-        ret = atom[0].eqNode(atom[1]);
-      } else {
-        unsigned one = 1;
-        BitVector bval(atom[0].getType().getConst<BitVectorSize>(), one);
-        Node bv_one = NodeManager::currentNM()->mkConst<BitVector>(bval);
-        ret = NodeManager::currentNM()
-                  ->mkNode(kind::BITVECTOR_PLUS, atom[0], bv_one)
-                  .eqNode(atom[1]);
-      }
+      unsigned one = 1;
+      BitVector bval(s.getType().getConst<BitVectorSize>(), one);
+      Node bv_one = nm->mkConst<BitVector>(bval);
```

Comment:
Use` bv::utils::mkOne(bv::utils::getSize(s))` to create a node representing bit-vector one.